### PR TITLE
INGM-412 Make the library compatible with Linux

### PR DESCRIPTION
**Pull request by pzinemanas**
### Description

In this PR fix some issues in order to have compatibility with Linux.

Fixes # INGM-412

### Type of change

Please add a description and delete options that are not relevant.

- [x] Fix how the network adapters are list in Linux
- [x] Raise exceptions when trying to run methods not supported in Linux.
- [x] Change pipeline to run no_connections tests in a Linux machine. 

### Tests
- [x] Fix tests
- [x] Run tests.

### Documentation

Please update the documentation.

- [x] Update docstrings of every function, method or class that change.
- [x] USe [type hints](https://docs.python.org/3/library/typing.html) for every function and argument.
- [x] Build documentation locally to verify changes.
- [ ] Add the changes at the `[Unreleased]` section of the [CHANGELOG](CHANGELOG.md).

### Code formatting

- [x] Use black package to format the code: `black -l 100 ingeniamotion tests`. It is recommended to configure the code editor to automatically format the code using black with a max length line of 100.

### CODE REVIEW

File: ingeniamotion/communication.py [hunk 1]
```diff
@@ -400,7 +412,10 @@ def get_ifname_by_index(index: int) -> str:
             IndexError: If interface index is out of range.
 
         """
-        return "\\Device\\NPF_{}".format(ifaddr.get_adapters()[index].name.decode("utf-8"))
+        if RUNNING_ON_WINDOWS:
```
Comment (MartinALL91) [inline ingeniamotion/communication.py:415]: This formatting is duplicated. I think it should be done in a single place.

File: ingeniamotion/communication.py [hunk 2]
```diff
@@ -31,6 +32,8 @@
 from ingeniamotion.comkit import create_comkit_dictionary
 from ingeniamotion.metaclass import DEFAULT_AXIS, DEFAULT_SERVO, MCMetaClass
 
+RUNNING_ON_WINDOWS = os.name == "nt"
```
Comment (MartinALL91) [inline ingeniamotion/communication.py:35]: Use https://docs.python.org/3/library/platform.html#platform.system

**Review by MartinALL91 — CHANGES_REQUESTED** [review]

File: ingeniamotion/communication.py (hunk 2 above)
Comment (pzinemanas) [reply ingeniamotion/communication.py]: Done [c2e9eb4](https://github.com/ingeniamc/ingeniamotion/pull/156/commits/c2e9eb442e82cead9cc83d945e4d3e119652c56a)

File: tests/test_communication.py [hunk 3]
```diff
@@ -272,7 +272,7 @@ def test_load_firmware_moco_exception(mocker, motion_controller):
 @pytest.mark.virtual
 def test_connect_servo_virtual():
     mc = MotionController()
-    mc.communication.connect_servo_virtual()
+    mc.communication.connect_servo_virtual(port=1062)
```
Comment (WoodElv) [inline tests/test_communication.py:275]: Why do you need to change the port?

**Review by WoodElv — APPROVED** [review]

File: tests/test_communication.py (hunk 3 above)
Comment (pzinemanas) [reply tests/test_communication.py:275]: Because the default port is already in use by a virtual drive created in the contftest.

**Review by MartinALL91 — APPROVED** [review]

**Review by GullinBustin — APPROVED** [review]

**Pull request merged**